tables/schema.py

```diff
@@ -179,11 +177,19 @@ def _handle_errors(self, schema_errors: pa.errors.SchemaErrors, table: pd.DataFr
                 continue
 
             if error_type == "column_in_schema":
-                # "column_in_schema" means that the table didn't fit the schema - dev error rather than user
+                # dataframe contains a column missing from the schema - dev error rather than user
                 raise TableExtractError(
                     f"Validated table contains column from outside of the schema - {failure.failure_case}"
                 )
 
+            if error_type == "column_in_dataframe":
```

Comment:
If we iterated over `SchemaErrors.schema_errors`, we would have access to `SchemaError` objects which have a `reason_code` property which contains enums from `funding-service-design-post-award-data-store/venv/lib/python3.11/site-packages/pandera/errors.py` such as `COLUMN_NOT_IN_SCHEMA` and `COLUMN_NOT_IN_DATAFRAME` which cover this case and the one above. Better to use constants where possible I think

Reply:
Looking into the `reason_code`s a bit further, it looks as if they're not granular enough for our use case. I can see that `coerce_dtype`, `dtype` and `not_nullable` all share the `SchemaErrorReason.SCHEMA_COMPONENT_CHECK` enum. I agree on your point about using constants though. I think we can just build our own enum for this use case and use that instead.

Reply:
Noted - I remember seeing this issue too. Each `SchemaError` object has a `check` string property which we could split on opening parenthesis as we currently do, which would allow us to use `reason_code` and the constants where they are available at the required specificity but the `check` string where not, but then if we're not removing the ugly splitting completely it's questionable whether this represents an actual improvement to the code. Shame that Pandera doesn't make it a bit easier for us here!

Anyway, this method is probably my least favourite part of the new validation code under `tables`, so any suggestions for improvements would be welcome, although don't need to fall under this PR given the bulk of the code is already in `main`. Would be good to hear / see more on the enums thing you mention there